src/httpClient/httpClient.ts

```diff
@@ -54,6 +56,7 @@ export default class HttpClient {
     this.correlationIdResolverFunction = options?.correlationIdResolver;
     this.enableCache = options?.enableCache ?? false;
     this.enableRetry = options?.enableRetry ?? false;
+    this.timeout = options?.timeout ?? 0;
```

Comment:
Can we not default it to `0` and leave it `undefined` instead? You're anyway doing check `if (this.timeout)` so `undefined` will work well. In that way we avoid any confusion with setting timeout to 0ms

Reply:
done